stablehlo/dialect/TypeInference.cpp

```diff
@@ -568,13 +568,20 @@ LogicalResult verifyReduceOpInputsAndInferShape(
 
 // Returns the types of the terminator arguments of the input  mlir::Block
 // 'block'.
-SmallVector<ShapedType> getAccumulatorTypes(Block& block) {
-  SmallVector<ShapedType> accumulatorSubShapes;
+LogicalResult getAccumulatorTypes(
```

Comment:
Optional nit: now that the function is not returning the accumulator types, consider renaming to `populateAccumulatorTypes`. This may or may not break places depending on how widely used this API is.

Reply:
The signature of the above function is same as `inferReduceOp` [ref](https://github.com/openxla/stablehlo/blob/d5b464925371092095ac934b46ba93ebd4284223/stablehlo/dialect/TypeInference.cpp#L2572) which takes a refernece to `infereedReturnShapes` for population within function. 

With that I propose to keep the function name as is.

Reply:
Why not return `FailureOr<SmallVector<ShapedType>>` ?

Reply:
I should take back my stance on that. done.